### PR TITLE
Feature/235 fly features

### PR DESCRIPTION
**Pull request by sambles**
* Added new features for complex model work flows

### CODE REVIEW

File: BFE_RShiny/oasisui/R/buildFly_module.R [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+    df_selectors <- result$tbl_modelsDetails$data_settings$datafile_selectors
+    lapply(seq_len(length(h)), function(i) {
+      check_ext <- paste0(".", unlist(df_selectors[[i]]$search_filters))
+      observeEvent(input[[h[i]]], {
+        for (j in seq_len(nrow(input[[h[i]]]))) {
+          if (paste0(".", tools::file_ext(input[[h[i]]][j,]$name)) %in% check_ext) {
+            .uploadDamageFile(file_entry = h[i], file_name = input[[h[i]]][j, ])
+          } else {
+            oasisuiNotification(type = "error", paste("Extension not supported."))
+          }
```
Comment (carlfischerjba) [inline BFE_RShiny/oasisui/R/buildFly_module.R:198]: I had some trouble defining a datafile selector in `model_settings.json` that would accept files with any extension. No combination of regular expressions or wildcards worked so I ended up hardcoding a fixed set of extensions. I think the issue is around here or possibly up on line 137.

I also note that the type of search filter is ignored, so `user`, `content_type` or `filename__contains` will all be treated as `filename__extension` (which isn't actually defined in the [schema](https://github.com/OasisLMF/OasisLMF/blob/master/oasislmf/schema/model_settings.json#L954)).

No pressure to fix this now.

**Review by carlfischerjba — COMMENTED** [review]
As mentioned just now to Sam, could we please remove references to "fly" and instead use a more generic term? I've heard the term "customisable model" and I think it appears on one of the buttons, although it's maybe not quite accurate since we're only configuring the analysis but doing it a bit earlier than usual.

Hard for me to say much about R but there are a number of commented out lines that should probably be deleted.

**Review by carlfischerjba — APPROVED** [review]
No objections now that some of the names have been made more generic. I'm not in a position to comment on anything else.

**Comment by sambles** [comment]
* Tested with PiWind using OasisEvaluation to check changes work with the older workflow (Looks good)
* Note: We need to document the changes and how to use the newer workflow, should be done before the next release

**Pull request merged**